src/RelayClient.ts

```diff
@@ -0,0 +1,130 @@
+import type { RelayMetadata } from '@rsksmart/rif-relay-common';
+import {
+  EnvelopingTypes,
+  IForwarder__factory,
+} from '@rsksmart/rif-relay-contracts';
+import config from 'config';
+import { constants, ethers } from 'ethers';
+import AccountManager from './AccountManager';
+import type { HubInfo, RelayOrDeployRequest } from './common/relay.types';
+import EnvelopingEventEmitter from './events/EnvelopingEventEmitter';
+
+type RelayOnlyRequestKey = keyof Omit<
+  EnvelopingTypes.RelayRequestStruct['request'],
+  keyof EnvelopingTypes.DeployRequestStruct['request']
+>;
+const relayOnlyRequestKeys: RelayOnlyRequestKey[] = ['gas'];
+
+type DeployOnlyRequestKey = keyof Omit<
+  EnvelopingTypes.DeployRequestStruct['request'],
+  keyof EnvelopingTypes.RelayRequestStruct['request']
+>;
+const deployOnlyRequestKeys: DeployOnlyRequestKey[] = ['index', 'recoverer'];
+
+const isRelayRequest = (
+  request: EnvelopingTypes.RelayRequestStruct
+): request is EnvelopingTypes.RelayRequestStruct => {
+  return relayOnlyRequestKeys.every((key: RelayOnlyRequestKey) => {
+    return key in request;
+  });
+};
+
+const isDeployRequest = (
+  request: EnvelopingTypes.DeployRequestStruct
+): request is EnvelopingTypes.DeployRequestStruct => {
+  return deployOnlyRequestKeys.every((key: DeployOnlyRequestKey) => {
+    return key in request;
+  });
+};
+
+type RequestConfig = {
+  isSmartWalletDeploy?: boolean;
+  smartWalletAddress?: string;
+  clientId?: string;
+  useEnveloping?: boolean;
+  forceGasPrice?: string;
+  forceGas?: string;
+  onlyPreferredRelays?: boolean;
+  ignoreTransactionReceipt?: boolean;
+  retries?: number;
+  initialBackoff?: number;
+};
+type PreparedRelayHttpRequest = {
+  relayRequest: RelayOrDeployRequest;
+  metadata: RelayMetadata;
+  requestConfig: RequestConfig;
+};
+
+class RelayClient extends EnvelopingEventEmitter {
+  constructor() {
+    super();
+  }
+
+  // TODO: lift the ts exception once method is used
+  // eslint-disable-next-line @typescript-eslint/ban-ts-comment
+  // @ts-ignore
+  private _prepareRelayHttpRequest = async (
+    { feesReceiver }: HubInfo,
+    request: EnvelopingTypes.RelayRequestStruct['request'],
+    data: EnvelopingTypes.RelayRequestStruct['relayData'],
+    requestConfig: RequestConfig
+  ): Promise<PreparedRelayHttpRequest> => {
+    const forwarderAddress =
+      config.get<string>('forwarderAddress') ??
+      data.callForwarder ??
+      constants.AddressZero;
+
+    const provider = ethers.getDefaultProvider();
```

Comment:
Where are we going to obtain or configure the RSK node address? Should we get it from the server or the configuration?